File: src/web/Bc.Web/BcWebEndpoint.cs

using Bc.Common.Endpoint;
using NServiceBus;
using NServiceBus.Persistence.Sql;

[assembly: SqlPersistenceSettings(MsSqlServerScripts = true)] 

namespace Bc.Web
{
    public static class BcWebEndpoint
    {
        public static EndpointConfiguration GetEndpoint()
        {
            const string endpointName = "Bc.WebEndpoint";
            
            var endpoint = EndpointCommon.GetEndpoint(
                endpointName,
                true,
                new EndpointCommonConfigurationProvider());

            return endpoint;
        }
    }
}

File: src/web/Bc.Web/BcWebEndpoint.fs
namespace Bc.Web

open Bc.Common.Endpoint
open Bc.Contracts.Internals.Endpoint.CommentTaking.Commands
open NServiceBus
open NServiceBus.Logging
open NServiceBus.Persistence.Sql
open Microsoft.Extensions.Hosting
open System

type BcWebEndpoint() =

    static member GetEndpoint() =
        let endpointName = "Bc.WebEndpoint"
        let destinationEndpointName = "Bc.Endpoint"

        let endpoint = getEndpoint endpointName true

        // logging
        let defaultFactory = LogManager.Use<DefaultFactory>()
        defaultFactory.Directory(AppDomain.CurrentDomain.BaseDirectory)

        // routing
        let setRouting (transport: TransportExtensions<'T>) =
           transport.Routing().RouteToEndpoint((typeof<TakeComment>).Assembly, destinationEndpointName)

        match ConfigurationProvider.isUseLearningTransportAndPersistence with
        | false -> setRouting (endpoint.UseTransport<SqlServerTransport>())
        | true -> setRouting (endpoint.UseTransport<LearningTransport>())

        endpoint